docs/site/DEVELOPING.md

```diff
@@ -286,6 +286,48 @@ LoopBack 4 documentation is hosted inside this monorepo in the
 This allows us to change both implementation and the documentation in a single
 pull request.
 
+### Organization of content
+
+LoopBack 4 is a large codebase organized into dozens of packages. We,
+developers, are often tempted to include a lot of details and wider context when
+writing docs. While this is good for capturing the knowledge in a text form, it
+also makes the documentation difficult to follow by people new to LoopBack.
+Newcomers get quickly overwhelmed by the amount of concepts they need to
+understand and all the different packages they need to know about.
```

Comment:
We should also mention the balance between `modular design` and `ease of use`.

- Module design leads to great flexibility and composability. As the number of modules grows, it also increases the complexity as developers need to know what to choose and how to use.
- We need to target high-level packages for application developers without dragging in lower-level details.

Reply:
We may want to identify several personas of different interest/entry to LB here:

- Platform/extension developers
- Application developers

Reply:
> We may want to identify several personas of different interest/entry to LB here:
>
> - Platform/extension developers
> - Application developers

Yeah, I agree that at some point, we need to step back and reorganize our docs to cater for different personas. As I see it, our current docs is mostly aimed at application developers. In this pull request, I am proposing how to make the docs easier to read for app developers. We can discuss how to better address other personas in https://github.com/strongloop/loopback-next/issues/5113.